packages/react-renderer-demo/src/app/pages/renderer/form-template.md

```diff
@@ -0,0 +1,70 @@
+import Grid from '@material-ui/core/Grid'
+import RawComponent from '@docs/raw-component';
+import RouterLink from 'next/link';
+import Link from '@material-ui/core/Link';
+
+import ListOfContents from '../../src/helpers/list-of-contents';
+
+<Grid container item>
+<Grid item xs={12} md={10}>
+
+# FormTemplate
+
+FormTemplate component allows you to fully customize the form. FormTemplate receives only two props:
+
+|Prop|Description|
+|----|-----------|
+|formFields|Fields of the form.|
+|schema|Schema from renderer. You can use it to extract a form title, description or whatever you need.|
+
+Minimal FormTemplate could look like this:
+
+```jsx
+const FormTemplate = ({schema, formFields}) => {
+  const { handleSubmit } = useFormApi();
+
+  return (
+    <form onSubmit={handleSubmit}>
+      { schema.title }
+      { formFields }
+      <button type="submit">Submit</button>
+    </form>
+  )
+}
+```
+
+With using `useFormApi` hook you can get access to all form information and functions you need. For example, you can use it to display form errors wherever you want to.
+
+## Customize form buttons
+
+You can customize form buttons in your <RouterLink href='/renderer/renderer-api#requiredprops'><Link href='/renderer/renderer-api#requiredprops'>FormTemplate</Link></RouterLink> component
+
+<RawComponent source="custom-buttons" />
+
+## Default FormTemplates
+
+FormTemplates of the provided DDF mappers offers additional customization via using props.
+
+```jsx
+import { FormTemplate } from ‘@data-driven-forms/pf4-component-mapper’
+
+<FormRenderer
+  ...
+  FormTemplate={(props) => <FormTemplate {props} showFormControls={false} ... />}
+/>
+```
+
+|Prop|Type|Description|Default|
+|----|:--:|----------:|------:|
+|buttonClassName|string|Class which will be given to the buttons wrapper.|{ }|
```

Comment:
This one is missing `onReset`, `onCancel` prop

Reply:
These are passed via formApi. They are still mentioned in the form renderer api.

Reply:
`onReset` and `reset` are different. You can extend the default `reset` function with `onReset`. Show some notification for instance.

Reply:
onReset and onCancel are passed to FormRenderer and the renderer is adding them to the context. And you access the context via useFormApi

Reply:
I know. I am just saying that I did not really notice it anywhere when going through the docs. Maybe this is a good candidate for example on a standalone page?

Reply:
onCancel and onReset all still mentioned in the renderer-api props list.

I don't think we need a standalone page for that, I can add it to custom-buttons example

Reply:
I know they are mentioned but it seems to be that it is buried under a lot of other props etc. An example of such a basic scenario is a must-have a feel.

Reply:
I added using of onCancel, onReset to custom-buttons.

However, I found that we should do two fixes:

However, I found that we should  enhance these function in the formRenderer context:

```jsx
onReset: () => { onReset && onReset(); reset(); }
onCancel: () => onCancel(values)
```

Reply:
Feel free to do them here.

Reply:
I would rather to do it in separate PR, this is out of the scope. :D